pgxn/neon/neon_walreader.c

```diff
@@ -254,7 +244,7 @@ NeonWALReadRemote(NeonWALReader *state, char *buf, XLogRecPtr startptr, Size cou
 
 					snprintf(start_repl_query, sizeof(start_repl_query),
 							 "START_REPLICATION PHYSICAL %X/%X (term='" UINT64_FORMAT "')",
-							 LSN_FORMAT_ARGS(startptr), state->wp->propTerm);
+                                                 LSN_FORMAT_ARGS(startptr), *(volatile uint64_t *)&GetWalpropShmemState()->mineLastElectedTerm);
```

Comment:
I'm not a fan of volatile, but without a specific reason, mostly because I haven't seen it's being used in postgres much.

WDYT about using mutexes for all shmem reads/writes?

Reply:
It seems to overkill to use mutex for updating a single 64-bit word at a time, especially when we have single writer, so I prefer to just use volatiles. It's also much less code.

Reply:
Agree. What about using `pg_atomic_read_u64/pg_atomic_write_u64` then? It looks like this is the preferred way to do atomic ops in postgres.

Reply:
Sure that works. pg_atomic_read/write is how I had it originally but changed it after discussion with Konstantin

Reply:
I'll keep `volatile` for `mineLastElectedTerm` because it is a read, and it's only used for logging. But I will switch `propEpochStartLsn` to atomic

Reply:
I don't understand, it seems better just to switch to `pg_atomic_read_u64` and `pg_atomic_write_u64` everywhere.

1. It's just `volatile` inside, at least on x86 :)
2. It explicitly marks all atomic operations

Reply:
Then you have to go change the type from `term_t` to `pg_atomic_u64` and make sure you cast it properly

Reply:
Hmm, I'm wondering how it's usually used in postgres codebase

Reply:
<img width="650" alt="image" src="https://github.com/neondatabase/neon/assets/8150127/893fe895-7c01-41fd-a33e-9cec4e04820c">

I don't see any issues with casting

Reply:
Switched to atomic